xarray/core/dataarray.py

```diff
@@ -4892,10 +4898,10 @@ def imag(self: T_DataArray) -> T_DataArray:
         return self._replace(self.variable.imag)
 
     def dot(
-        self: T_DataArray,
+        self,
         other: T_DataArray,
         dims: Dims = None,
-    ) -> T_DataArray:
+    ) -> Self:
```

Comment:
Again other needs to be changed

Reply:
Good spot — found a few of these. I suspect I haven't quite got them all though.

V happy to add a few more if you spot them. I suspect the only way to catch em all is with tests though...